lldb/source/Plugins/ScriptInterpreter/Python/ScriptInterpreterPython.cpp

```diff
@@ -134,9 +115,30 @@ struct InitializePythonRAII {
       PyImport_AppendInittab("_lldb", LLDBSwigPyInit);
     }
 
+#if LLDB_EMBED_PYTHON_HOME
```

Comment:
Was moving the initialization past the `if (!Py_IsInitialized()) {` needed ?

Reply:
Part of it, yes, but you could avoid changing the order by splitting the ifdef-ed code in two, which is slightly messier. I enabled `LLDB_EMBED_PYTHON_HOME` on Darwin and there it doesn't matter. That would look like this:

```
#if LLDB_EMBED_PYTHON_HOME
  PyConfig config;
  [...]
  PyConfig_SetBytesString(&config, &config.home, g_python_home.c_str());
#endif 

  if (!Py_IsInitialized()) {
  	[...]
    PyImport_AppendInittab("_lldb", LLDBSwigPyInit);
  }
  
#if LLDB_EMBED_PYTHON_HOME  
    config.install_signal_handlers = 0;
    Py_InitializeFromConfig(&config);
    PyConfig_Clear(&config);
#else
    Py_InitializeEx(/*install_sigs=*/0);
#endif
```